language/move-prover/bytecode/src/compositional_analysis.rs

```diff
@@ -89,12 +91,21 @@ where
             for exit_state in exit_states.iter().skip(1) {
                 acc.join(&exit_state);
             }
-            data.annotations.set(acc)
+            data.annotations.set(map(acc))
         } else {
             // TODO: not clear that this is desired, but some clients rely on
             // every function having a summary, even natives
-            data.annotations.set(initial_state)
+            data.annotations.set(map(initial_state))
         }
         data
     }
+
+    fn summarize(
```

Comment:
- Nice solution to the client boilerplate problem
- This is a pre-existing issue, but would you mind adding doc comments to explain the difference between `summarize` and `summarize_map`?

Reply:
Thanks! I took a stab at the doc. I moved a lot of the text from the comment describing the trait and placed it above the functions `summarize` and (now) `summarize_abstract` themselves. Not sure if this is desirable/if the comments should be in both places. Please let me know what best practice is here.

Reply:
Actually I think it would be more idiomatic to make abstract a trait function. It's likely possible to solve the boilerplate problem with a default implementation. (See comment above.)

Having a function parameter for `abstract` opens the possibility to use for the same instance of the analysis different versions of abstract, which is odd. Also, you may run into trouble accessing state of the impl of CompositionalAnalysis.